### PR TITLE
Solve MCMCChains break change

### LINKED ISSUE #76: MCMCChains breaking change cause `MCMCChains.Chains` convert failed
`MCMCChains v4.0.0` introduces a break change, in which `Symbol` is used as variable names instead of `String`. See:

https://github.com/TuringLang/MCMCChains.jl/pull/197

https://github.com/TuringLang/MCMCChains.jl/issues/185

So now, related Turing code in quickstart will fail. Another MWE:

```julia
    using MCMCChains
    using ArviZ

    # Define the experiment
    n_iter = 500
    n_name = 3
    n_chain = 2

    # experiment results
    val = randn(n_iter, n_name, n_chain) .+ [1, 2, 3]'
    val = hcat(val, rand(1:2, n_iter, 1, n_chain))

    # construct a Chains object
    chn = MCMCChains.Chains(val)  # According to version, this may introduce String or Symbol name

    ArviZ.summary(chn)
```

A PR will follow.

### PR DESCRIPTION
**Pull request by yiyuezhuo**
fix #76

### CODE REVIEW

**Comment by yiyuezhuo** [comment]
CI failed on `ValueError('Number of rows must be a positive integer, not 0')`, since `fig, _ = plt.subplots(0, 0, figsize=figsize, **backend_kwargs)` is invalid in current matplotlib version. It's a `ArviZ` bug and was fixed in recent commit:

https://github.com/arviz-devs/arviz/commit/29d4b13969a97c49584f103b3419feade9dea8bf

So [Job 363.4](https://travis-ci.com/github/arviz-devs/ArviZ.jl/jobs/366662593) passed, because it set `ARVIZ_VERSION=master` to use newest ArviZ code. I think that `ArviZ.jl` has nothing to do.

**Comment by ahartikainen** [comment]
can you downgrade your matplotlib to 3.2?

    pip install "matplotlib==3.2"

**Comment by yiyuezhuo** [comment]
If specified `ARVIZ_VERSION = 'master'`, `python -m pip install git+https://github.com/pydata/xarray git+https://github.com/arviz-devs/arviz` will be used to install `arviz`:

https://github.com/arviz-devs/ArviZ.jl/blob/master/.travis.yml#L56

Otherwise, it will install latest conda-forge version:

https://github.com/arviz-devs/ArviZ.jl/blob/master/src/ArviZ.jl#L96

**Comment by sethaxen** [comment]
Thanks for the PR! ArviZ.jl has gotten a little out-of-sync with ArviZ due to my busy-ness (see #75 and #79). I'll open a few PRs to fix these errors/warnings then come back and review this one.

**Comment by yiyuezhuo** [comment]
@sethaxen , `Turing@14.0` have been released, which use `MCMCChains@4.0.4`, so my MVE will fail as expected (in the release version) and this PR become useful I guess.

File: src/mcmcchains.jl [hunk 1]
```diff
@@ -61,8 +61,10 @@ attributes_dict(::Nothing) = Dict()
 
 function section_dict(chns::Chains, section)
     ndraws, _, nchains = size(chns)
-    loc_names = string.(getfield(chns.name_map, section))
-    vars_to_locs = varnames_locs_dict(loc_names)
+    loc_names_old = getfield(chns.name_map, section) # old may be Symbol or String
+    loc_names = string.(loc_names_old)
+    loc_str_to_old = Dict(name_str => name_old for (name_str, name_old) in zip(loc_names, loc_names_old))
```
Comment (sethaxen) [inline src/mcmcchains.jl:66]: Suggestion from the new format checking action (https://github.com/arviz-devs/ArviZ.jl/pull/77/checks?check_run_id=1076393897#step:5:29)
```suggestion
    loc_str_to_old =
        Dict(name_str => name_old for (name_str, name_old) in zip(loc_names, loc_names_old))
```

**Review by sethaxen — CHANGES_REQUESTED** [review]
Thanks for your patience! The approach looks right to me, and the test suite passes with Turing v0.13 and MCMCChains v3. I'm happy to approve once the following are handled:
- Bump compat of MCMCChains in `Project.toml` to v4 (I will bump docs compats in a subsequent PR) and test suite passes
- Format checker passes (see suggestion)
- Increment patch version number

File: Project.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 CmdStan = "5.2.3, 6.0"
 Conda = "1.0"
 DataFrames = "0.20, 0.21"
-MCMCChains = "0.3.15, 0.4.2, 1.0, 2.0, 3.0"
+MCMCChains = "0.3.15, 0.4, 1.0, 2.0, 3.0"
```
Comment (sethaxen) [inline Project.toml:21]: This should do it
```suggestion
MCMCChains = "0.3.15, 0.4, 1.0, 2.0, 3.0, 4.0"
```

Comment (yiyuezhuo) [reply Project.toml:21]: I misread `0.4` as `4.0` :(

**Comment by sethaxen** [comment]
Looks like MCMCChains v4 also removed the `sorted` keyword from `Chains(...)` (see https://github.com/TuringLang/MCMCChains.jl/pull/197)

**Comment by yiyuezhuo** [comment]
> Looks like MCMCChains v4 also removed the `sorted` keyword from `Chains(...)` (see [TuringLang/MCMCChains.jl#197](https://github.com/TuringLang/MCMCChains.jl/pull/197))

It seems like that I used a wrong version to run local unit tests...

**Review by sethaxen — APPROVED** [review]
LGTM! Thanks again!

**Pull request merged**